index.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+const util = require('util');
+const Promise = require('bluebird');
+
+// Custom mysql dialect with deadlock retries
+const Client_MySQL = require('knex/lib/dialects/mysql');
+
+class Client_MySQL_deadlock extends Client_MySQL {
+  _query(connection, obj) {
+    const self = this;
+    const args = arguments;
+    let retryAmount = obj.retryAmount || 3;
+
+    function runQuery() {
+      return Reflect.apply(Client_MySQL.prototype._query, self, args)
```

Comment:
In this case `super` won't work, because it can be used only inside methods. `runQuery()` is a plain function, defined inside a method. So I'd stick with `Reflect.apply` instead.